File: Algorithm/baekjoon/bj_1124.py
N,M = map(int,input().split())

isprime = [True] * (M+1)
isprime[0]=isprime[1]=False
primes=[]
for i in range(2,M+1):
    if isprime[i]:
        primes.append(i)
        for j in range(i*i,M+1,i):
            isprime[j] = False

prime_cnt = [0]*(M+1)

for prime in primes:
    for i in range(prime,M+1,prime):
        tmp = i
        while not tmp%prime:
            prime_cnt[i]+=1
            tmp //=prime

ans = 0
for num in range(N,M+1):
    if isprime[prime_cnt[num]]:
        ans += 1
print(ans)

File: Algorithm/baekjoon/java/src/bj_1124.java
import java.io.*;
import java.util.*;


public class bj_1124 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        // N,M
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());

        // isPrime
        boolean[] isPrime = new boolean[M+1];
        for (int i=0; i<M+1; i++) {
            isPrime[i] = true;
        }
        isPrime[0]=isPrime[1]=false;

        // primes
        ArrayList<Integer> primes = new ArrayList<>();
        for (int i=2; i<M+1; i++) {
            if (isPrime[i]) {
                primes.add(i);
                for (int j=i*2; j<M+1; j+=i) {
                    isPrime[j] = false;
                }
            }
        }
        // primeCnt
        int[] primeCnt = new int[M+1];

        for (int prime : primes) {
            for (int i=prime; i<M+1; i+=prime) {
                int tmp = i ;
                while (tmp%prime==0) {
                    primeCnt[i] ++ ;
                    tmp /= prime ;
                }
            }
        }
        int ans = 0 ;
        for (int i=N; i<M+1; i++) {
            if (isPrime[primeCnt[i]]) ans ++;
        }

        System.out.println(ans);
    }
}
